format/spec.md

```diff
@@ -496,6 +496,7 @@ A snapshot consists of the following fields:
 | _optional_ |            | **`manifests`**          | A list of manifest file locations. Must be omitted if `manifest-list` is present |
 | _optional_ | _required_ | **`summary`**            | A string map that summarizes the snapshot changes, including `operation` (see below) |
 | _optional_ | _optional_ | **`schema-id`**          | ID of the table's current schema when the snapshot was created |
+|            | _optional_ | **`statistics`**         | A list of [statistics files' metadata](#statistics-file). The field should be retained by writers, unless writer updates the statistics, or knows they became obsolete. |
```

Comment:
Why would there be more than one stats file? Wouldn't multiple files be merged together or one would make another obsolete?

Reply:
We don't have implementation for merging stat files yet.
One application can calculate some statistic on some columns and then someone else may can calculate a different statistic (or on different list of columns). This would lead to multiple stat files.

Reply:
Trying to understand this (sorry if missed previous discussion), does a puffin file or list of puffin files have statistics for all data referenced by a snapshot?  We will not support statistics per data file?

Reply:
Trying to understand this (sorry if missed previous discussion), does a puffin file or list of puffin files have statistics for all data referenced by a snapshot?  We will not support statistics per data file?

Reply:
> We don't have implementation for merging stat files yet.

I think we want to make spec decisions based on what makes the most sense long term, not on what we've already built. It should be easy enough to merge these. I think we should have just one statistics file.

Reply:
That means the transaction adding new statistics will need to rewrite the existing ones in the commit phase.
That would also imply different API -- it will no longer make sense for the stats writing application to write the stats file (and then commit its registration). Instead the API should take statistics in-memory, because their file destination is to be determined.
I am not yet convinced these are fair tradeoffs.

Reply:
I looked at the other PR, but I don't follow the argument here that it would make more sense to pass stats in memory. Whether you base stats on the previous file or not doesn't matter to the commit phase. The process updating stats produces a file and then sets it as the current one on a snapshot.

Reply:
I asked for further explanation in the other PR (https://github.com/apache/iceberg/pull/4741#discussion_r895458863)